week-four/app/src/test/java/pro/marvinhosea/movielist/ui/movies/MovieViewModelTest.kt

```diff
@@ -1,37 +1,43 @@
 package pro.marvinhosea.movielist.ui.movies
 
-import kotlinx.coroutines.GlobalScope
-import kotlinx.coroutines.launch
+import com.nhaarman.mockito_kotlin.mock
+import com.nhaarman.mockito_kotlin.verify
+import org.junit.After
 import org.junit.Test
 
-import org.junit.Assert.*
-import pro.marvinhosea.movielist.data.models.Movie
-import pro.marvinhosea.movielist.data.models.POPULAR_CATEGORY
+import org.junit.Before
+import org.koin.core.context.startKoin
+import org.koin.core.context.stopKoin
+import org.koin.test.KoinTest
+import org.koin.test.get
+import org.mockito.MockitoAnnotations
+import pro.marvinhosea.movielist.di.networkModule
+import pro.marvinhosea.movielist.di.repositoryModule
+import pro.marvinhosea.movielist.di.viewModelModule
+import pro.marvinhosea.movielist.repository.MoviesRepository
 
-class MovieViewModelTest {
-    private val modeMovieViewModel by lazy { MovieViewModel() }
+class MovieViewModelTest : KoinTest {
+    private val moviesRepository: MoviesRepository = mock()
+    private lateinit var movieViewModel: MovieViewModel
 
-    @Test
-    fun testAllMoviesHasObservers() {
-        GlobalScope.launch {
-            val movies = modeMovieViewModel.getAllMovies()
-            assertTrue(movies.hasObservers())
+    @Before
+    fun setUp() {
+        startKoin{
+            modules(listOf(networkModule, repositoryModule, viewModelModule))
         }
+        MockitoAnnotations.initMocks(this)
+        movieViewModel = MovieViewModel(get())
```

Comment:
Why do you mock the MoviesRepository and then use Koin to inject in tests?

You should always use mocks in tests, and not use DI which relies on Context, and the Android application, _**which you don't have in tests**_.